Website/src/rendering/src/components/Checkout/CheckoutSummary.tsx

```diff
@@ -55,7 +56,7 @@ const CheckoutSummary = (props: CheckoutSummaryProps): JSX.Element => {
     if (!payments?.length || !payments[0] || !payments[0].ID || !payments[0].Accepted) {
       return false;
     }
-    if (isAnonymous && order?.FromUser?.Email === 'test@test.com') {
+    if (isAnonymous && !props.guestUserEmail) {
```

Comment:
Would that return false in OrderReviewDetails.tsx if we do not pass the email from there?

Would it be best for CheckoutSummary to accept an optional `shouldEnableButton?: () => boolean` prop. If the prop is defined, we could call the function from here and return false if it returns false. Then we could move the `isAnonymous && !props.guestUserEmail` logic in the `CheckoutDetails.tsx` component only. That would not impact other places we use `CheckoutSummary`.